res/res_pjsip_logger.c

```diff
@@ -664,16 +684,35 @@ static void check_debug(void)
 {
 	RAII_VAR(char *, debug, ast_sip_get_debug(), ast_free);
 
+	/* Got directive to disable debug */
 	if (ast_false(debug)) {
-		pjsip_disable_logger(-1);
+		/* If the logger was enabled via the CLI instead of through the config file,
+		 * then we shouldn't disable it on a reload.
+		 * Only disable logging if logging isn't enabled via the CLI. */
+		if (logger_cli_settings == AST_PJSIP_LOGGER_NONE) {
+			/* Logger not enabled via CLI currently so good to go ahead and disable. */
+			pjsip_disable_logger(-1);
+		} else {
+			ast_debug(3, "Leaving logger enabled since logging settings overridden using CLI\n");
+		}
+		logger_config_settings = AST_PJSIP_LOGGER_NONE;
 		return;
 	}
 
+	/* Got directive to enable debug */
 	if (ast_true(debug)) {
-		pjsip_enable_logger_all(-1);
+		/* If logger already enabled via config, then nothing has changed. */
+		if (!(logger_config_settings & AST_PJSIP_LOGGER_ALL)) {
```

Comment:
The cli setting isn't considered here, and the config is set to NONE in the ast_false case either way so I'm not sure what this check or the conf setting enum is for.
If a cli no should over-ride the config yes, then I think this needs to mirror the ast_false case.  Otherwise I don't think this check or the config enum is needed.